### PR TITLE
add Spring boot distro version

### PR DESCRIPTION
**Pull request by zeitlinger**
~~and minor bug fixes:~~
~~- [add @ConditionalOnMissingBean to LoggingMetricExporter](https://github.com/open-telemetry/opentelemetry-java-instrumentation/commit/faa262fe7359e1bc922fec848cf1779c643e9fad) -> https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10283~~
~~- [add LoggingMetricExporterAutoConfiguration to spring factories](https://github.com/open-telemetry/opentelemetry-java-instrumentation/commit/dd6b4b20d0a78ae6dd0e0c95386e4e3cfb1f0a72) -> https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10282~~

### CODE REVIEW

File: instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/resources/DistroVersionResourceProvider.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.instrumentation.api.internal.EmbeddedInstrumentationProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.sdk.resources.Resource;
+
+@AutoService(ResourceProvider.class)
```
Comment (laurit) [inline instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/resources/DistroVersionResourceProvider.java:16]: do you really need the `@AutoService` here?

Comment (zeitlinger) [reply instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/resources/DistroVersionResourceProvider.java]: no, good catch

**Review by jeanbisutti — COMMENTED** [review]
Could you add tests as it is done with LoggingSpanExporter?

**Comment by jeanbisutti** [comment]
Could you add tests as it is done with `LoggingSpanExporter`?

> Could you add tests as it is done with LoggingSpanExporter?

Tests on `LoggingMetricExporter`

**Comment by jeanbisutti** [comment]
> and minor bug fixes:
> 
> * [add @ConditionalOnMissingBean to LoggingMetricExporter](https://github.com/open-telemetry/opentelemetry-java-instrumentation/commit/faa262fe7359e1bc922fec848cf1779c643e9fad)
> * [add LoggingMetricExporterAutoConfiguration to spring factories](https://github.com/open-telemetry/opentelemetry-java-instrumentation/commit/dd6b4b20d0a78ae6dd0e0c95386e4e3cfb1f0a72)

Two issues, two PRs? This could make it easier to track changes.

**Comment by zeitlinger** [comment]
> > and minor bug fixes:
> > 
> > * [add @ConditionalOnMissingBean to LoggingMetricExporter](https://github.com/open-telemetry/opentelemetry-java-instrumentation/commit/faa262fe7359e1bc922fec848cf1779c643e9fad)
> > * [add LoggingMetricExporterAutoConfiguration to spring factories](https://github.com/open-telemetry/opentelemetry-java-instrumentation/commit/dd6b4b20d0a78ae6dd0e0c95386e4e3cfb1f0a72)
> 
> Two issues, two PRs? This could make it easier to track changes.

done

**Comment by zeitlinger** [comment]
> > Could you add tests as it is done with LoggingSpanExporter?
> 
> Tests on `LoggingMetricExporter`

has this been resolved by splitting the PR?

File: instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/resources/DistroVersionResourceProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                      .createResource(DefaultConfigProperties.createFromMap(ImmutableMap.of()))
+                      .getAttributes()
+                      .asMap())
+              .containsEntry(
+                  AttributeKey.stringKey("telemetry.distro.name"),
+                  "opentelemetry-spring-boot-starter")
+              .anySatisfy(
+                  (key, val) -> {
+                    assertThat(key.getKey()).isEqualTo("telemetry.distro.version");
+                    assertThat(val).asString().matches("\\d\\.\\d\\.\\d-alpha-SNAPSHOT");
```
Comment (laurit) [inline instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/resources/DistroVersionResourceProviderTest.java:44]: please ensure that this won't fail for release versions

Comment (zeitlinger) [reply instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/resources/DistroVersionResourceProviderTest.java]: @laurit done

**Review by laurit — APPROVED** [review]

**Review by trask — APPROVED** [review]

**Pull request merged**